debvader/deblender.py

```diff
@@ -97,60 +97,42 @@ def get_residual_field(self, res_deblend=None):
         deblended_image = self.field_image.copy()
 
         if res_deblend is not None:
+
             for isolated_galaxy_row in res_deblend:
 
                 # TODO: may be use some inbuild padding function?
-
                 # subtract the image of the currently galaxy after positioning it correctly
                 # TODO: try to avoid using scipy.ndimage.shift here, it is super slow. Diectly subtract at correct index instead of creating a padding?
 
                 pos_offset = int((self.field_size - self.cutout_size) / 2)
+                predicted_galaxy = isolated_galaxy_row["output_images_mean"].copy()
 
                 if self.optimize_positions:
 
-                    # First create padded images of the stamps at the size of the field to allow for a simple subtraction.
-
-                    output_images_mean_padded = np.zeros(
-                        (self.field_size, self.field_size, self.nb_of_bands)
-                    )
-
-                    output_images_mean_padded[
-                        pos_offset : self.cutout_size + pos_offset,
-                        pos_offset : self.cutout_size + pos_offset,
-                        :,
-                    ] = isolated_galaxy_row["output_images_mean"]
-
-                    x_pos = (
-                        isolated_galaxy_row["galaxy_distances_to_center_x"]
-                        + isolated_galaxy_row["shifts"][0]
-                    )
-                    y_pos = (
-                        isolated_galaxy_row["galaxy_distances_to_center_y"]
-                        + isolated_galaxy_row["shifts"][1]
-                    )
+                    x_shift = isolated_galaxy_row["shifts"][0]
+                    y_shift = isolated_galaxy_row["shifts"][1]
 
                     for band in range(self.nb_of_bands):
-                        deblended_image[:, :, band] -= scipy.ndimage.shift(
-                            output_images_mean_padded[0, :, :, band],
-                            shift=(x_pos, y_pos),
-                        )
-                else:
-                    x_start = int(
-                        round(
-                            isolated_galaxy_row["galaxy_distances_to_center_x"]
-                            + pos_offset
+                        predicted_galaxy[:, :, band] = scipy.ndimage.shift(
+                            predicted_galaxy[:, :, band],
+                            shift=(x_shift, y_shift),
```

Comment:
In the case of integer shifts, if you want to avoid using `scipy.ndimage.shift` and stick to numpy, a way would be a combination of `np.pad` to create the padding around the image and then a `np.roll` in each direction to shift the central image.

Reply:
this **may** be a way to solve #61

Reply:
I think the idea of having the shift was to correct for the pixelization error so it is always a float value